tools/hapcut2/hapcut2.xml

```diff
@@ -0,0 +1,135 @@
+<tool id="hapcut2" name="Hapcut2" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@+ga@GA_VERSION_SUFFIX@">
+  <xrefs>
+    <xref type="bio.tools">hapcut2</xref>
+  </xrefs>
+  <macros>
+    <token name="@TOOL_VERSION@">1.3.3</token>
+    <token name="@VERSION_SUFFIX@">0</token>
+    <token name="@GA_VERSION_SUFFIX@">1</token>
+  </macros>
+  <requirements>
+    <requirement type="package" version="1.3.3"> hapcut2 </requirement>
+  </requirements>
+
+  <command detect_errors="exit_code"><![CDATA[
+
+## Prep inputs
+## =====================================================================
+ln -s '$input_bam' input.bam &&
+ln -s '$input_vcf' input.vcf &&
+
+
+## Run program
+## =====================================================================
+## Extract variant fragments from alignment
+extractHAIRS --bam input.bam --VCF input.vcf --out frags.dat
+#if $optimization == 'pacbio':
+  --pacbio 1
+#elif $optimization == 'ont':
+  --ont 1
+#elif $optimization == 'hic':
+  --HiC 1
+#end if
+&&
+
+## Create haplotype.out and haplotype.out.phased.VCF
+HAPCUT2 --fragments frags.dat --VCF input.vcf --output haplotype.out
+#if $optimization == 'hic':
+  --HiC 1
+#end if
+&&
+
+
+## Collect outputs
+## =====================================================================
+#if $output_fragments:
+    cp frags.dat '$frags' &&
```

Comment:
i think it's better to pick this up in the `<output>` element so you're not writing to the output directory

Reply:
Normally I'd agree but this tool is a bit weird in that it "magically" creates outputs based on the `haplotype.out` filename.
When you say `--output haplotype.out` it will also create `haplotype.out.phased.VCF` - which gets really confusing when you try to collect the latter output. So I thought it was just clearer here to create verbose outputs and let Galaxy clean up the wdir after completion (which apparently happens quickly anyway).

Reply:
ok sounds good. so i know in my wrappers, what happens if you add 

`from_work_dir="haplotype.out.phased.VCF"` to the phased haplotype block you have here:

```xml
    <data name="haplotype_phased" format="vcf" label="${tool.name} on ${on_string}: Phased haplotype block">
```